swarmrl/engine/espresso.py

```diff
@@ -612,15 +651,26 @@ def integrate(self, n_slices, force_model: swarmrl.models.InteractionModel = Non
         """
 
         if not self.integration_initialised:
+            self.slice_idx = 0
+            self.step_idx = 0
             self._setup_interactions()
             self._remove_overlap()
             self._init_h5_output()
+            # print("write_interval", self.params.write_interval.m_as("sim_time") )
+            # print("time_slice", self.params.time_slice.m_as("sim_time") )
+            # print("time_step", self.params.time_step.m_as("sim_time") )
+            # print("write idx, slice idx, step idx",self.write_idx, self.slice_idx,
+            # self.step_idx)
+            # print("n_slices",n_slices)
             self.integration_initialised = True
 
-        for _ in range(n_slices):
+        old_slice_idx = self.slice_idx
```

Comment:
Because I don't know that to each loop iteration also belongs a slice iteration. This is the case in the current code and also if the write_ interval is separated to the slice_ time by a multiplicative factor.